### PR TITLE
Upgrade dependencies

### LINKED ISSUE #76: Update tokio when reqwest updates it

### LINKED ISSUE #101: Upgrading to Tokio v1
I've recently been working on cratetorrent, and it's really great. The only gripe I have is that it uses Tokio v0.2, instead of Tokio v1, making it incompatible with a project I'm working on.

### PR DESCRIPTION
**Pull request by pawanjay176**
Fixes #76 
Fixes #101 

Upgrade all dependencies (except `bitvec`). 

some not straightforward updates:
1. No need to use fused futures with `UnboundedReceiver` .
2. length reading code in `codec.rs` changed to use some extra apis provided in bytes 1.0 (tests work, but not sure if code can be improved further).

### CODE REVIEW

File: cratetorrent/src/peer/codec.rs
Comment (jbrady42) [inline cratetorrent/src/peer/codec.rs:409]: Perhaps using `std::io::Cursor` as described in https://tokio.rs/tokio/tutorial/framing#parsing to keep the `get_*` API would be more straightforward for these peek ahead checks.

```
let mut tmp_buf = Cursor::new(&buf);
let msg_len = tmp_buf.get_u32() as usize;
```

File: cratetorrent-cli/Cargo.toml
Comment (mandreyel) [inline cratetorrent-cli/Cargo.toml:19]: I think we can just use the major version. If `tokio` follows SemVer, there shouldn't be backwards compatibility breakage with minor updates.

File: cratetorrent/src/peer/codec.rs
Comment (mandreyel) [reply cratetorrent/src/peer/codec.rs:409]: Agreed, this looks more straightforward. Also, keeping a brief comment on why the temporary buffer would be useful for future maintainers.

**Review by mandreyel — CHANGES_REQUESTED** [review]
Ah, I planned to work on this first thing I came back to this project. I'm glad you did it though, thanks for picking this up!

A few small things.

I compiled your branch @pawanjay176, and got the following warnings:

```rust
   Compiling cratetorrent v0.1.0 (/tmp/cratetorrent/cratetorrent)
warning: variant is never constructed: `InvalidPieceIndex`
  --> cratetorrent/src/disk/error.rs:44:5
   |
44 |     InvalidPieceIndex,
   |     ^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

warning: variant is never constructed: `InvalidPieceIndex`
  --> cratetorrent/src/disk/error.rs:65:5
   |
65 |     InvalidPieceIndex,
   |     ^^^^^^^^^^^^^^^^^

warning: 2 warnings emitted
```

Then, we should bump the minor version to `0.2.0` to indicate breakage for downstream consumers due to major updates to dependencies. Both for the crate and the CLI.

**Comment by mandreyel** [comment]
What was the issue with `bitvec`, by the way?

**Comment by pawanjay176** [comment]
@mandreyel Sorry for fixing the review comments so late.

> got the following warnings

These warnings are present in the master branch as well, so I didn't touch those. Can fix those up in this PR if you want though :)

> What was the issue with bitvec, by the way?

The bitvec changes were a little more involved as the upstream api had changed. I felt it would be better to have only tokio changes in this PR.

**Comment by mandreyel** [comment]
Thanks for addressing the comments!

> These warnings are present in the master branch as well, so I didn't touch those. Can fix those up in this PR if you want though :)

You're right, my bad. They are out of scope for this MR, I am fixing them separately. Thanks though.

> The bitvec changes were a little more involved as the upstream api had changed. I felt it would be better to have only tokio changes in this PR.

Fair enough, @jbrady42 has submitted a patch separately that I will apply on top of yours once done.
---

There is one more thing before we can merge. The binary used for the integration tests seems not to have been updated for latest tokio, based on this error:

```
error[E0599]: the method `next` exists for struct `tokio::sync::mpsc::unbounded::UnboundedReceiver<cratetorrent::alert::Alert>`, but its trait bounds were not satisfied
  --> src/main.rs:72:38
   |
72 |     while let Some(alert) = alert_rx.next().await {
   |                                      ^^^^ method cannot be called on `tokio::sync::mpsc::unbounded::UnboundedReceiver<cratetorrent::alert::Alert>` due to unsatisfied trait bounds
   | 
```

If you don't have time we can merge and I'll send a fixup MR, but if you could fix this that would be fantastic.

Thanks for your work.

**Comment by carbotaniuman** [comment]
Any chance at getting this and #105 merged? I'm looking to make some updates and PRs to this crate, but I don't want to retread past work.

**Comment by mandreyel** [comment]
@carbotaniuman Yes, sorry, I was dealing with personal issues. I'm merging these MRs and sending a fixup to finish this work. Thanks @pawanjay176 & @jbrady42.

**Pull request merged**
